### ISSUE TITLE
Allow usage of __init_subclass__ in Models

### ISSUE THREAD

**Issue by conqp** [coleifer/peewee#2651]
Currently, one cannot use `__init_subclass__()` on `Model`s.
This is useful, if one wants to create an abstract model to be implemented with different parameters.
Unfortunately `ModelBase` currently hands off all kwargs to the parent type `type` resulting in `TypeError`s:

```python
from peewee import Model


class MyTemplate(Model):

    def __init_subclass__(cls, *args, some_param: str, **kwargs):
        super().__init_subclass__(*args, **kwargs)

        if some_param is not None:
            cls.SOME_PARAM = some_param


class MyImplementation(MyTemplate, some_param='hello world'):
    pass
```
Resulting in:
```shell
$ python3 test.py
Traceback (most recent call last):
  File "/root/test.py", line 13, in <module>
    class MyImplementation(MyTemplate, some_param='hello world'):
TypeError: __new__() got an unexpected keyword argument 'some_param'
```

**Comment by conqp** [comment]
Darn it. This is already fixed in `3.15.4`. I was using `3.14.8` in that particular machine.
Sorry for the noise.

**Comment by coleifer** [comment]
No problem at all, I only just released that fix, with thanks to @smac89 for the patch.
